addons/mail/models/mail_thread.py

```diff
@@ -3587,13 +3588,17 @@ def _notify_by_email_prepare_rendering_context(self, message, msg_vals=False,
             'record_name': record_name,
             'subtitles': [record_name],
             # user / environment
+            'author_user': author_user,  # User who sends the message
             'company': company,
             'email_add_signature': email_add_signature,
             'lang': lang,
             'signature': signature,
             'website_url': website_url,
             # tools
             'is_html_empty': is_html_empty,
+            # display
+            'email_show_header': self.env.context.get('email_show_header', False),
```

Comment:
All in all, this could be something like
```suggestion
            'email_show_footer': self.env.context.get('force_email_footer') or (has_button_access and author_user),
            'email_show_header': self.env.context.get('force_email_header') or (has_button_access or subtitles),
```